File: python/bubble_sort.py


def bubble_sort(array):
    list_size = len(array)
    # loop through the list n times
    for count in list(range(list_size)):
        # go throught the list until the sorted part
        # swap if the element found is greater than the next element
        for item in list(range(list_size - count - 1)):
            if array[item] > array[item + 1]:
                array[item], array[item +1 ] = array[item + 1], array[item]
        
    return array

# Driver code to test above
arr = [64, 34, 25, 12, 22, 11, 90]
print(bubble_sort(arr))

File: javascript/bubblesort.js
function bubblesort(numbers_array){
    // go through the array n times
    for (i = 0;  i < numbers_array.length; i++){
        // go through the array until the sorted parent
        // if item found is greater that the next item swap them
        for (j = 0; j < numbers_array.length - i - 1; j++){
            if (numbers_array[j] > numbers_array[j+1]){
                const temp = numbers_array[j]
                numbers_array[j] =  numbers_array[j+1]
                numbers_array[j+1] = temp 
            }
        }
    }
    return numbers_array
}

numbers = [64, 34, 25, 12, 22, 11, 90]
console.log(bubblesort(numbers))